addons/test_mass_mailing/tests/test_mailing.py

```diff
@@ -117,6 +117,7 @@ def test_mailing_gateway_update(self):
         mailing.invalidate_recordset()
         self.assertMailingStatistics(mailing, expected=5, delivered=4, sent=5, opened=1, clicked=1, bounced=1)
         self.assertEqual(recipients[1].message_bounce, 1)
+        self.assertEqual(mailing.mailing_trace_ids[1].failure_reason, 'This is the bounce email')
```

Comment:
Mmmh ... can't we do a ``assertMailTraces`` and the right mailing trace and support failure_reason in it ? This assert seems a bit bug-prone and hard to debug when it fails.